protocol/0029-FEES-fees.md

```diff
@@ -184,9 +184,29 @@ For example, Ether is 18 decimals (wei). The smallest unit, non divisible is 1 w
+- In auction mode, if the price taker has insufficient asset to cover the total fee in their general + margin account, then the shortfall should be ignored, the orders should remain (instead of being rejected)(<a name="0029-FEES-041" hre1f="#0029-FEES-041">0029-FEES-041</a>)
+
+- When there is `high_volume_market_maker_rebate`, `high_volume_maker_fee` should be taken from the `treasury/buyback_fee` components with value `high_volume_maker_fee = high_volume_factor * trade_value_for_fee_purposes`(<a name="0029-FEES-042" href="#0029-FEES-042">0029-FEES-042</a>)
+- When there is `high_volume_market_maker_rebate`, treasury fee will be updated to: `treasury_fee = treasury_fee * (1 - high_volume_maker_fee / (treasury_fee + buyback_fee))`(<a name="0029-FEES-043" href="#0029-FEES-043">0029-FEES-043</a>)
+- When there is `high_volume_market_maker_rebate`, buyback fee will be updated to: `buyback_fee = buyback_fee * (1 - high_volume_maker_fee / (treasury_fee + buyback_fee))`(<a name="0029-FEES-044" href="#0029-FEES-044">0029-FEES-044</a>)
+
+- Once total fee is collected, `infrastructure_fee = infrastructure_fee + maker_fee + liquidity_fee + buyback_fee + treasury_fee` is transferred to infrastructure fee pool for that asset at the end of fee distribution time. (<a name="0029-FEES-045" href="#0029-FEES-045">0029-FEES-045</a>)
+- Once total fee is collected, `maker_fee = fee_factor[maker]  * trade_value_for_fee_purposes` is transferred to maker at the end of fee distribution time. (<a name="0029-FEES-046" href="#0029-FEES-046">0029-FEES-046</a>)
+- Once total fee is collected, the `high_volume_maker_fee` is transferred to maker at the end of fee distribution time. (<a name="0029-FEES-047" href="#0029-FEES-047">0029-FEES-047</a>)
+- Once total fee is collected, `liquidity_fee = fee_factor[liquidity] * trade_value_for_fee_purposes` is distributed to liquidity providers as described in [this spec](./0042-LIQF-setting_fees_and_rewarding_lps.md).(<a name="0029-FEES-048" href="#0029-FEES-048">0029-FEES-048</a>)
+- Once total fee is collected, `treasury_fee = fee_factor[treasury] * trade_value_for_fee_purposes` (with `high_volume_maker_fee` deducted) is transferred to the treasury fee pool for that asset, where it will remain until community governance votes for transfers.(<a name="0029-FEES-049" href="#0029-FEES-049">0029-FEES-049</a>)
+- Once total fee is collected, `buyback_fee = fee_factor[buyback] * trade_value_for_fee_purposes` (with `high_volume_maker_fee` deducted) is transferred to the buyback fee pool for that asset, where it will remain until community governance votes for transfers or a regular purchase program is set up.(<a name="0029-FEES-050" href="#0029-FEES-050">0029-FEES-050</a>)
```

Comment:
```suggestion
- Once total fee is collected, `buyback_fee = fee_factor[buyback] * trade_value_for_fee_purposes` (with with apropriate fraction of `high_volume_maker_fee` deducted) is transferred to the buyback fee pool for that asset, where it will remain until community governance votes for transfers or a regular purchase program is set up.(<a name="0029-FEES-050" href="#0029-FEES-050">0029-FEES-050</a>)
```